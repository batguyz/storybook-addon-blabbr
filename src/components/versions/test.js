import React from 'react';
import renderer from 'react-test-renderer'; // eslint-disable-line
import Versions from './';

test('Versions dropdown not shown when no versions required (null)', () => {
  const tree = renderer.create(
    <Versions />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Versions dropdown not shown when no versions required (empty)', () => {
  const tree = renderer.create(
    <Versions versions={[]} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Versions dropdown shows the supplied versions', () => {
  const tree = renderer.create(
    <Versions versions={[
      '0.0.1',
      '0.0.2',
      '0.0.3',
    ]}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

