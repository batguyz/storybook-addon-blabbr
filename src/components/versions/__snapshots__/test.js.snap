// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Versions dropdown not shown when no versions required (empty) 1`] = `null`;

exports[`Versions dropdown not shown when no versions required (null) 1`] = `null`;

exports[`Versions dropdown shows the supplied versions 1`] = `
<div
  id="blabbr-version-history"
>
  <div
    className="dropdown"
  >
    <button
      className="dropbtn"
    >
      0.3.6
    </button>
    <div
      className="dropdown-content"
    >
      <a
        href="about://:/0.0.3/"
        target="_parent"
      >
        0.0.3
      </a>
      <a
        href="about://:/0.0.2/"
        target="_parent"
      >
        0.0.2
      </a>
      <a
        href="about://:/0.0.1/"
        target="_parent"
      >
        0.0.1
      </a>
    </div>
  </div>
</div>
`;
